subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,1930 @@
+                assert integTestRuntimeClasspathFileNames.containsAll('commons-lang3-3.11.jar', 'guava-29.0-jre.jar', 'mysql-connector-java-6.0.6.jar')
+                assert !integTestRuntimeClasspathFileNames.contains('servlet-api-2.5.jar'): 'compileOnly dependency'
+            }
+        }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        desc                | implementationNotationTest                     | compileOnlyNotationTest                              | runtimeOnlyNotationTest                        | implementationNotationInteg                     | compileOnlyNotationInteg                              | runtimeOnlyNotationInteg
+        'GAV string'        | gavStr(guavaGroup, guavaName, guavaVerTest)    | gavStr(servletGroup, servletName, servletVerTest)    | gavStr(mysqlGroup, mysqlName, mysqlVerTest)    | gavStr(guavaGroup, guavaName, guavaVerInteg)    | gavStr(servletGroup, servletName, servletVerInteg)    | gavStr(mysqlGroup, mysqlName, mysqlVerInteg)
+        'GAV map'           | gavMap(guavaGroup, guavaName, guavaVerTest)    | gavMap(servletGroup, servletName, servletVerTest)    | gavMap(mysqlGroup, mysqlName, mysqlVerTest)    | gavMap(guavaGroup, guavaName, guavaVerInteg)    | gavMap(servletGroup, servletName, servletVerInteg)    | gavMap(mysqlGroup, mysqlName, mysqlVerInteg)
+        'named args'        | namedArgs(guavaGroup, guavaName, guavaVerTest) | namedArgs(servletGroup, servletName, servletVerTest) | namedArgs(mysqlGroup, mysqlName, mysqlVerTest) | namedArgs(guavaGroup, guavaName, guavaVerInteg) | namedArgs(servletGroup, servletName, servletVerInteg) | namedArgs(mysqlGroup, mysqlName, mysqlVerInteg)
+    }
+
+    def 'can add dependencies to the implementation, compileOnly and runtimeOnly configurations of a suite via DependencyHandler using #desc'() {
+        given:
+        buildFile << """
+        plugins {
+          id 'java-library'
+        }
+
+        ${mavenCentralRepository()}
+
+        testing {
+            suites {
+                integTest(JvmTestSuite)
+            }
+        }
+
+        dependencies {
+            // production code requires commons-lang3 at runtime, which will leak into tests' runtime classpaths
+            implementation 'org.apache.commons:commons-lang3:3.11'
+
+            testImplementation $implementationNotationTest
+            testCompileOnly $compileOnlyNotationTest
+            testRuntimeOnly $runtimeOnlyNotationTest
+
+            // intentionally setting lower versions of the same dependencies on the `test` suite to show that no conflict resolution should be taking place
+            integTestImplementation project
+            integTestImplementation $implementationNotationInteg
+            integTestCompileOnly  $compileOnlyNotationInteg
+            integTestRuntimeOnly $runtimeOnlyNotationInteg
```

Comment:
I don't think we care about testing this. We only care about the test suite dependencies block

Reply:
This was one of the [existing tests](https://github.com/gradle/gradle/blob/da9db6394bb4f51c413b23c1380c51de755ade3c/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesDependenciesIntegrationTest.groovy#L144-L208).  I don't think there's any harm in leaving it?